src/components/add-products/add-products.js

```diff
@@ -1,38 +1,58 @@
 import React, { useState } from 'react';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
 
+import { productActions } from '../../store/products/action';
 import BuildForm from '../form-builder';
 import ProductCard from './product-card';
-import { productActions } from '../../store/products/action';
 
 import './product.scss';
 
 function AddProducts() {
+  const id = Number(useParams().id);
+  const products = useSelector(({ products }) => products.products);
+  let product = null;
+  let completedForm = false;
   const dispatch = useDispatch();
   const [submitted, setSubmitted] = useState(false);
+
+  if (id !== 0 && products) {
+    product = products.find((object) => object.id === id);
+  }
+
   const [inputs, setInputs] = useState({
-    title: '',
-    price: '',
-    categoryId: 'Bread',
-    imageUrl: '',
+    title: product ? product.title : '',
+    price: product ? product.price : '',
+    categoryId: product ? product.categoryId : 'Bread',
+    imageUrl: product ? product.imageUrl : '',
   });
 
+  const { title, price, categoryId, imageUrl } = inputs;
   const onChange = (event) => {
     const { name, value } = event.target;
     setInputs((inputs) => ({ ...inputs, [name]: value }));
   };
 
+  if (title && price && categoryId && imageUrl) {
+    completedForm = true;
+  }
+
   const onSubmit = (event) => {
     setSubmitted(true);
-    if (title && price && categoryId && imageUrl) {
-      dispatch(productActions.addProduct(title, price, categoryId, imageUrl));
-      dispatch(productActions.getProducts());
-    }
     event.preventDefault();
+    if (completedForm) {
+      if (id === 0) {
+        dispatch(productActions.addProduct(title, price, categoryId, imageUrl));
+        dispatch(productActions.getProducts());
```

Comment:
You can move this to the root of `if (completedForm) {...}` Since it is a followup event to adding/editing that occurs in both the `if` and the `else` with regardless of the condition of `id` it doesn't need to be apart of the conditions concerning `id`